### PR TITLE
New option --no-color to disable colored output

### PR DESCRIPTION
**Pull request by siebert**
New option --no-color to disable color. Useful for build system logs and consoles without color support.

### CODE REVIEW

File: gitlint/__init__.py [hunk 1]
```diff
@@ -202,6 +199,16 @@ def main(argv, stdout=sys.stdout, stderr=sys.stderr):
 
     vcs, repository_root = get_vcs_root()
 
+    no_color = arguments['--no-color']
+    if no_color:
```
[the remark this answers is not in the document]
Comment (siebert) [reply gitlint/__init__.py]: I'm ok with this and the next request

File: gitlint/__init__.py [hunk 2]
```diff
@@ -202,6 +199,16 @@ def main(argv, stdout=sys.stdout, stderr=sys.stderr):
 
     vcs, repository_root = get_vcs_root()
 
+    no_color = arguments['--no-color']
```
[the remark this answers is not in the document]
Comment (siebert) [reply gitlint/__init__.py]: So you would like to have `color = not arguments['--no-color']`?

**Comment by siebert** [comment]
Thanks for the prompt review. I thought the reason for this change is obvious, especially as it is already on the list of TODOS... I want to use git-lint in a shell job on Jenkins and the color codes are rather distracting in the output, so I want the option to disable them.

Regarding the restructuring of the output formatters, I'm not sure if that additional effort is something I'd like to make for this change. Would you consider to merge my patch without it? Would it be ok to define the new classes within __init__.py or must they reside in separate modules?

File: gitlint/__init__.py [hunk 3]
```diff
@@ -202,6 +199,16 @@ def main(argv, stdout=sys.stdout, stderr=sys.stderr):
 
     vcs, repository_root = get_vcs_root()
 
+    error = 'ERROR'
+    skipped = 'SKIPPED'
+    okay = 'OK'
+
+    color = not arguments['--no-color']
```
Comment (sk-) [inline gitlint/__init__.py]: Could you change this to:

```python
color = not arguments['--no-color'] and sys.stdout.isatty()
```

Comment (siebert) [reply gitlint/__init__.py]: I'll close this PR and reopen it with a dedicated branch so other PRs are no longer in it.
I'll also adjust the TODO, but I don't really see how to add a useful test for it.
The git lint command is just called in a shell script by Jenkins. Jenkins only evaluates the return code, but if something goes wrong the developer consults the console output of the Jenkins run and that was rather hard to read with the color codes in it.

**Pull request closed without merging**